### DISCUSSION TITLE
CVE-2024-45519

### DISCUSSION THREAD

**Discussion by demogorgonz** [ianw1974/zimbra-build-scripts#96] [General]
https://www.bleepingcomputer.com/news/security/critical-zimbra-rce-flaw-exploited-to-backdoor-servers-using-emails/

Seems 10.1.0 is affected:

> According to [Zimbra's security bulletin](https://wiki.zimbra.com/wiki/Zimbra_Security_Advisories), CVE-2024-45519 has been resolved in version 9.0.0 Patch 41 or later, versions 10.0.9 and 10.1.1, and Zimbra 8.8.15 Patch 46 or later.

Any plans to release 10.1.1 ?

**Comment by ianw1974** [comment]
They were built yesterday from the develop branch so anything committed to develop branch should be in the builds irrespective of the version number.  That said I can redo the builds.  This does of course only guarantee that it will be patched **if** Zimbra made the commits to the develop branch.  One of the CVE's from last year they didn't commit to develop until 3 weeks later.

**Reply by jeredfloyd** [reply]
This article breaks down the actual vulnerability: https://blog.projectdiscovery.io/zimbra-remote-code-execution/

I just got out of a meeting and am going to poke at what cases is actually applies.  It's about postjournal which seems to be a compliance feature so I'm not sure it even affects FOSS builds.

**Reply by demogorgonz** [reply]
This seems to be POC : https://github.com/p33d/CVE-2024-45519 , just tried and seems version 10.1.0 is not affected, would love if someone could double check it.

**Reply by jeredfloyd** [reply]
I do not believe this affects FOSS builds as the vulnerable binary `/opt/zimbra/libexec/postjournal` is not included.

**Reply by demogorgonz** [reply]
> I do not believe this affects FOSS builds as the vulnerable binary `/opt/zimbra/libexec/postjournal` is not included.

You are right!

**Reply by ianw1974** [reply]
Also:

```
[zimbra@zimbra10 root]$ zmlocalconfig -s | grep -i journal
postjournal_archive_bounce_to = <>
postjournal_archive_rcpt_to = <>
postjournal_enabled = false
postjournal_helo_name = localhost
postjournal_per_user_journaling = 1
postjournal_smtp_read_timeout = 60
postjournal_strip_postfix_proxy = 1
sqlite_journal_mode = PERSIST
```

it's disabled: ```false```.  Now whether they can enable this remotely or not is another matter.  Perhaps the CVE only works if postjournal was enabled and running.

**Comment by jeredfloyd** [comment]
OK, the vulnerability here is an a feature -- Exchange-like Journaling - that appears to be:
1) Network edition only
2) Disabled by default

It was introduced in Zimbra 8.5 and must be enabled explicitly as described here https://wiki.zimbra.com/wiki/New_Features_ZCS_8.5

The vulnerable binary `/opt/zimbra/libexec/postjournal` is not present on my system and the package for it does not seem to appear in the FOSS repos.

**Reply by ianw1974** [reply]
Glad to know OSE/FOSS is all clear irrespective of the version.  At least until some other CVE's are found that may be present in both versions.
